FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_light_weight/DemoTasks/LightWeightMQTTExample.c

```diff
@@ -710,10 +810,60 @@ static void prvMQTTSubscribeToTopic( Socket_t xMQTTSocket )
                              ( void * ) xBuffer.pBuffer,
                              xPacketSize,
                              0 );
+
     configASSERT( xStatus == ( BaseType_t ) xPacketSize );
 }
 /*-----------------------------------------------------------*/
 
+static void prvMQTTSubscribeWithBackoffRetries( Socket_t xMQTTSocket )
+{
+    RetryUtilsStatus_t xRetryUtilsStatus = RetryUtilsSuccess;
+    RetryUtilsParams_t xRetryParams;
+
+    /* Initialize retry attempts and interval. */
+    xRetryParams.maxRetryAttempts = MAX_RETRY_ATTEMPTS;
+    RetryUtils_ParamsReset( &xRetryParams );
+
+    do
+    {
+        /* The client is now connected to the broker. Subscribe to the topic
+         * as specified in mqttexampleTOPIC at the top of this file by sending a
+         * subscribe packet then waiting for a subscribe acknowledgment (SUBACK).
+         * This client will then publish to the same topic it subscribed to, so it
+         * will expect all the messages it sends to the broker to be sent back to it
+         * from the broker. This demo uses QOS0 in Subscribe, therefore, the Publish
+         * messages received from the broker will have QOS0. */
+        LogInfo( ( "Attempt to subscribe to the MQTT topic %s.\r\n", mqttexampleTOPIC ) );
+        vTaskDelay( pdMS_TO_TICKS( 3000 ) );
```

Comment:
why is this delay required? If it is required, can you please make it  MACRO and add a comment about the reason for the delay?

Reply:
whoops that was just something I left in to test if it's working